src/util/findRouteByComponent.js

```diff
@@ -1,24 +1,23 @@
-export default function findRouteByComponent(component, routes) {
+export default function findRouteByComponent(component, routes, components) {
   const result = {};
-  for (const route of routes) {
-    if (route.component === component) {
-      result.route = route;
-      return result;
-    }
-    if (route.components) {
-      const foundNamedComponent = Object.keys(route.components)
+  components.some((matchedComponent, i) => {
+    if (typeof matchedComponent === 'object') {
```

Comment:
We can probably use `lodash.isplainobject` here since we have it as a dependency already.